src/Token.php

```diff
@@ -20,11 +22,57 @@ class Token
 
     /**
      * @param string $value
-     * @param string $type
      */
-    public function __construct(string $value, string $type)
+    public function __construct(string $value)
     {
         $this->value = $value;
-        $this->type = $type;
+        $this->type = $this->getTokenType();
+    }
+
+    /**
+     * Returns StripeToken type base on incoming token.
+     *
+     * @return string
+     */
+    private function getTokenType(): string
+    {
+        switch (true) {
+            case $this->startsWith('pi'):
+                return StripeToken::PAYMENT_INTENT;
+            case $this->startsWith('pm'):
+                return StripeToken::PAYMENT_METHOD;
+            case $this->startsWith('cus'):
+                return StripeToken::CUSTOMER;
+            case $this->startsWith('ch'):
+                return StripeToken::CHARGE;
+            default:
+                return StripeToken::UNDEFINED;
+        }
+    }
+
+    /**
+     * Checks chars from the beginning of the token.
+     *
+     * @param string $needle
+     * @param string|null $token
+     *
+     * @return bool
+     */
+    private function startsWith(string $needle, string $token = null): bool
+    {
+        $length = strlen($needle);
+        return (substr(trim($token ?? $this->value), 0, $length) === $needle);
+    }
+
+    /**
+     * Identifies card token.
+     *
+     * @param string $token
+     *
+     * @return bool
+     */
+    public function isCard(string $token): bool
+    {
+        return $this->startsWith('card', $token);
     }
```

Comment:
not keen on this, would just have `$token->type === Token::TYPE_CARD` or whatever